opto/trainer/algorithms/basic_algorithms.py

```diff
@@ -136,13 +103,8 @@ def train(self,
                 backup_dict = {p: copy.deepcopy(p.data) for p in self.agent.parameters()}
 
                 # Forward the agent on the inputs and compute the feedback using the guide
-                if use_asyncio: # Run forward asynchronously
-                    outputs = async_run([self.forward]*len(xs),
-                                       [(self.agent, x, guide, info) for x, info in zip(xs, infos)],
-                                       max_workers=num_threads,
-                                       description=f"Forward pass (batch size: {len(xs)})")  # async forward
-                else: # Run forward sequentially
-                    outputs = [self.forward(self.agent, x, guide, info) for x, info in zip(xs, infos) ]
+                forward = batch_run(max_workers=num_threads, description=f"Forward pass (batch size: {len(xs)})")(self.forward)
```

Comment:
With this kind of syntax, I think we should call it `batchify` or `enable_async_batch` or `batch_asyncify` -- I strongly oppose `batch_run` because you are not running this function on this line, you are running this function on the next line. It might be confusing to users.